### DISCUSSION TITLE
MongoDB ObjectID conversion in dove

### DISCUSSION THREAD

**Discussion by lukashass** [fratzinger/feathers-casl#87] [Q&A]
I recently started converting an app using dove `pre-17` to `5.0.0`. This brought along the switch to `@feathersjs/mongodb` and with it the conversion of strings to ObjectIDs [in a resolver](https://feathersjs.com/api/databases/mongodb.html#resolveobjectid) instead of in mongoose.

Since feathers-casl calls `_get` internally, the conversion is skipped. My first idea was to use `resolveQueryObjectId` manually when defining my rules e.g.: `can('read', 'users', { _id: resolveQueryObjectId({ $in: userIdsManagedByUser }) });`
This breaks the ability in the frontend since IDs are strings in the frontend.

Is this something we can address in feathers-casl or do you have any better ideas to solve this problem?

Greetings from Kiel :)

**Comment by fratzinger** [comment]
Moin.

I don't use NoSQL at all so I don't have the same problem. There are multiple possibilities:
1. just use `get` instead of `_get` which maybe has performance downsides
2. introduce 'hooks' to `casl` explicitely. Something like:
```ts
authorize({
  adapter: 'mongodb',
  hooks: {
    around: { ... }
  }
})
```
3. Take the rules on the frontend and transform the objectIds to strings. <- (this is maybe the quickest solution for your problem as of now)
